lib/matplotlib/tests/test_sphinxext.py

```diff
@@ -12,11 +12,11 @@
 
 
 def test_tinypages(tmpdir):
-    html_dir = pjoin(str(tmpdir), 'html')
-    doctree_dir = pjoin(str(tmpdir), 'doctrees')
+    html_dir = tmpdir.join('html')
```

Comment:
There's a `tmp_path` fixture in pytest nowadays, which gives you a `Path`. It's better to use `Path` everywhere and not mess with `py.path.local` from `tmpdir` anymore.

Reply:
I this works as is I'm ok with defering the switch to `tmp_path` to a later PR.

Reply:
It does work, but this PR switches from plain string paths to py.path.local. If we switch we should directly go to Path.

Reply:
We don't require new enough pytest to use that, though maybe we should.

Reply:
I propose to bump the pytest dependency then. `tmp_path` is much simpler than `tmpdir`. We currently require pytest>=3.6. Let's go with `pytest>=5` (4.x had a number of tmp_path fixes), which is their python 3.6+-only release and is now 9 months old. IMHO that's reasonable for a testing dependency.

Reply:
I rewrote this part so it's as close to using `tmp_path` as it can.

Pytest 5 is in neither Debian nor Fedora, so that should probably be a different discussion.

Reply:
ok.